lib/ansible/module_utils/k8s_common.py

```diff
@@ -0,0 +1,290 @@
+                    if option == 'choices':
+                        if isinstance(option_value, dict):
+                            spec[arg_name]['choices'] = [value for key, value in option_value.items()]
+                        else:
+                            spec[arg_name]['choices'] = option_value
+                    else:
+                        spec[arg_name][option] = option_value
+
+        self.argspec_cache = spec
+        return self.argspec_cache
+
+    def execute_module(self):
+        """
+        Performs basic CRUD operations on the model object. Ends by calling
+        AnsibleModule.fail_json(), if an error is encountered, otherwise
+        AnsibleModule.exit_json() with a dict containing:
+          changed: boolean
+          api_version: the API version
+          <kind>: a dict representing the object's state
+        :return: None
+        """
+
+        if self.params.get('debug'):
+            self.helper.enable_debug(reset_logfile=False)
+            self.helper.log_argspec()
+
+        resource_definition = self.params.get('resource_definition')
+        if self.params.get('src'):
+            resource_definition = self.load_resource_definition(self.params['src'])
+        if resource_definition:
+            resource_params = self.resource_to_parameters(resource_definition)
+            self.params.update(resource_params)
+
+        state = self.params.get('state', None)
+        name = self.params.get('name')
+        namespace = self.params.get('namespace', None)
+        existing = None
+
+        return_attributes = dict(changed=False, api_version=self.api_version)
+        return_attributes[self.helper.base_model_name_snake] = {}
+
+        try:
+            auth_options = {}
+            for key, value in self.helper.argspec.items():
+                if value.get('auth_option') and self.params.get(key) is not None:
+                    auth_options[key] = self.params[key]
+            self.helper.set_client_config(**auth_options)
+        except OpenShiftException as e:
+            self.fail_json(msg='Error loading config', error=str(e))
+
+        if state is None:
+            # This is a list, rollback or ? module with no 'state' param
+            if self.helper.base_model_name_snake.endswith('list'):
+                # For list modules, execute a GET, and exit
+                k8s_obj = self._read(name, namespace)
+                return_attributes[self.kind] = k8s_obj.to_dict()
+                self.exit_json(**return_attributes)
+            elif self.helper.has_method('create'):
+                # For a rollback, execute a POST, and exit
+                k8s_obj = self.__create(namespace)
+                return_attributes[self.kind] = k8s_obj.to_dict()
+                return_attributes['changed'] = True
+                self.exit_json(**return_attributes)
+            else:
+                self.fail_json(msg="Missing state parameter. Expected one of: present, absent, replaced.")
```

Comment:
replaced is not a good state, it should be part of present (would be 'changed' return status)